akka-actor-typed-tests/src/test/scala/docs/akka/typed/InteractionPatternsSpec.scala

```diff
@@ -189,4 +215,64 @@ class InteractionPatternsSpec extends TestKit with TypedAkkaSpecWithShutdown {
     probe.expectNoMessage(1.millisecond)
     probe.expectMsg(2.seconds, Batch(Vector[Msg](ExcitingMessage("one"), ExcitingMessage("two"))))
   }
+
+  "contain a sample for ask" in {
+    // #actor-ask
+    trait HalProtocol
+    case class OpenThePodBayDoorsPlease(respondTo: ActorRef[HalResponse]) extends HalProtocol
+    case class HalResponse(message: String)
+
+    val halBehavior = Behaviors.immutable[HalProtocol] { (ctx, msg) ⇒
+
+      msg match {
+        case OpenThePodBayDoorsPlease(respondTo) ⇒
+          respondTo ! HalResponse("I'm sorry Dave, I cannot do that!")
+          Behaviors.same
+      }
+    }
+
+    trait DaveProtocol
+    // this is a part of the protocol that is internal to the actor itself
+    case class AdaptedResponse(message: String) extends DaveProtocol
+
+    def daveBehavior(hal: ActorRef[HalProtocol]) = Behaviors.deferred[DaveProtocol] { ctx ⇒
+
+      // asking someone requires a timeout, if the timeout hits without response
+      // the ask is failed with a TimeoutException
+      implicit val timeout: Timeout = 3.seconds
+
+      // Note: The second parameter list takes a function ActorRef[T] => Message,
+      // as OpenThePodBayDoorsPlease is a case class it has a factory apply method
+      // that is what we are passing as the second parameter here it could also be written
+      // OpenThePodBayDoorsPlease.apply or ref => OpenThePodBayDoorsPlease(ref)
```

Comment:
I'm not sure `OpenThePodBayDoorsPlease.apply` is useful to mention here, but `ref => OpenThePodBayDoorsPlease(ref)` certainly is!